pyccel/ast/core.py

```diff
@@ -248,27 +247,46 @@ class AsName(Basic):
 
     Parameters
     ==========
-    name   : str
-             original name of variable or function
+    obj    : Basic or BasicType
+             The variable, function, or module being renamed
     target : str
              name of variable or function in this context
     """
-    __slots__ = ('_name', '_target')
+    __slots__ = ('_obj', '_target')
     _attribute_nodes = ()
 
-    def __init__(self, name, target):
-        self._name = name
+    def __init__(self, obj, target):
+        if PyccelAstNode.stage != "syntactic":
+            assert (isinstance(obj, Basic) and \
+                    not isinstance(obj, PyccelSymbol)) or \
+                   (isinstance(obj, type) and \
+                   Basic in obj.__mro__)
```

Comment:
Is there an advantage compared to `issubclass(obj, Basic)`?

Reply:
I believe the difference is that `Basic in obj.__mro__` is equivalent to : `issubclass(obj, Basic) or isinstance(obj, Basic)`
However this isn't needed here, the only reason that I didn't use `issubclass` is because it didn't show up when I googled how to do this ^^'

Reply:
Fixed 0a810e67